.github/actions/deploy-s3-app/action.yml

```diff
@@ -0,0 +1,67 @@
+name: Deploy frontend to S3
+description: Deploys static files to AWS S3
+
+inputs:
+  aws-access-key-id:
+    required: true
+    description: AWS access key ID
+  aws-secret-access-key:
+    required: true
+    description: AWS secret access key
+  aws-region:
+    required: true
+    description: AWS region
+
+  aws-s3-bucket-name:
+    required: true
+    description: AWS S3 bucket name
+  graasp-app-id:
+    required: false
+    default: ''
+    description: App Id for deploying to the apps bucket. Do not specify to deploy to the root of the bucket.
+  version:
+    required: false
+    default: 'latest'
+    description: The version of the app that will be deployed
```

Comment:
I don't think it makes sense to "graaspify" this workflow. Since these inputs are only used to define the bucket path, why not make it so that you can pass an optional root path instead? That way, you could deploy things other than apps and frontends (e.g. it would be useful for H5P integration).

Reply:
@dialexo  I see what you mean, and I think it makes sens to try to make this composite workflow as general as possible. My concern is that it would shift the responsibility over to the consumer of the workflow, and this means that for an app for example we would not have a good way to enforce the pattern of where apps are deployed since the consumer can define the root path they want. We should maybe discuss and see if there are some rules we can define in the buckets to not allow to push to the root of the bucket for example. @pyphilia

Reply:
I understand. What about creating another composite workflow to deploy apps specifically, which takes the required inputs, and then feeds into this generic composite workflow?

Reply:
Yes I guess this would be a good idea, I just need to check if it is allowed to create composite that calls a composite 🤷